Crypto Wars/Assets/Scripts/TileScript.cs

```diff
@@ -1,48 +1,72 @@
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine;
 
 public class TileScript : MonoBehaviour
 {
-    public bool isPlayerControlled;
-    public bool isEnemyControlled;
-    public int boardXPos;
-    public int boardYPos;
+    // Backing fields
+    private bool _isPlayerControlled;
+    private bool _isEnemyControlled;
+    
+    // Public properties
+    public int BoardXPos { get; private set; }
+    public int BoardYPos { get; private set; }
+
+    // Properties for player and enemy control with encapsulated logic to update the tile color
+    public bool IsPlayerControlled
+    {
+        get { return _isPlayerControlled; }
+        set
+        {
+            _isPlayerControlled = value;
+            UpdateTileColor();
+        }
+    }
+
+    public bool IsEnemyControlled
+    {
+        get { return _isEnemyControlled; }
+        set
+        {
+            _isEnemyControlled = value;
+            UpdateTileColor();
+        }
+    }
 
+    // References to the renderer and materials for the tile
     private MeshRenderer rendererReference;
-    private MeshFilter filterReference;
     private Material playerColor;
     private Material enemyColor;
 
-    // Start is called before the first frame update
+    // Initialization in Start method
     void Start()
     {
         rendererReference = GetComponent<MeshRenderer>();
-        playerColor = Resources.Load("Materials/PlayerTileColor", typeof(Material)) as Material;
-        enemyColor = Resources.Load("Materials/EnemyTileColor", typeof(Material)) as Material;
-        
-        SetStartingColor();
-    }
-
-    // Update is called once per frame
-    void Update()
-    {
-        
+        playerColor = Resources.Load<Material>("Materials/PlayerTileColor");
+        enemyColor = Resources.Load<Material>("Materials/EnemyTileColor");
```

Comment:
enemyColor and playerColor need casting, otherwise they produce a compiler error